src/contracts/types/Ownable.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity 0.8.9;
+
+interface IOwnable {
+    function manager() external view returns (address);
+
+    function renounceManagement() external;
+
+    function pushManagement(address newOwner_) external;
+
+    function pullManagement() external;
+}
+
+contract Ownable is IOwnable {
+    address internal _owner;
+    address internal _newOwner;
+
+    event OwnershipPushed(
+        address indexed previousOwner,
+        address indexed newOwner
+    );
+    event OwnershipPulled(
+        address indexed previousOwner,
+        address indexed newOwner
+    );
+
+    constructor() {
+        _owner = msg.sender;
+        emit OwnershipPushed(address(0), _owner);
+    }
+
+    function manager() public view override returns (address) {
```

Comment:
One more thing we should clean up - 

Don't understand the "owner" vs "manager" syntax here. We should fix that so its either all "manager" or all "owner".

Reply:
This was already the implementation, but I agree it was weird because when i first saw manager, it sounded like there could be multiple managers.  I'll change to owner because that makes sense